contracts/eden/src/eden-micro-chain.cpp

```diff
@@ -1662,6 +1662,94 @@ void resign(eosio::name account)
    remove_if_exists<by_pk>(db.members, account);
 }
 
+void rename(eosio::name old_account, eosio::name new_account)
+{
+   auto update = [&](auto& acc) {
+      if (acc == old_account)
+         acc = new_account;
+   };
+   auto update_vec = [&](auto& vec) {
+      std::replace(vec.begin(), vec.end(), old_account, new_account);
+   };
+
+   db.status.modify(get_status(), [&](auto& status) { update_vec(status.status.initialMembers); });
+
+   if (auto* obj = get_ptr<by_pk>(db.balances, old_account))
+      db.balances.modify(*obj, [&](auto& obj) { obj.account = new_account; });
+
+   for (auto& obj : db.balance_history)
+      db.balance_history.modify(obj, [&](auto& obj) {
+         update(obj.account);
+         update(obj.other_account);
```

Comment:
It's a bit unfortunate that other_account forces us to iterate over the whole table.